catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStorePublisher.java

```diff
@@ -88,79 +114,69 @@ public void setUnavailable() {
-                String localRegistryId = registryIdentityMetacard.getAttribute(
-                        RegistryObjectMetacardType.REGISTRY_ID)
-                        .getValue()
-                        .toString();
-                Security.runAsAdminWithException(() -> {
-                    registryPublicationService.publish(localRegistryId,
-                            registryStore.getRegistryId());
-                    return null;
-                });
+            StringBuilder logMessage = new StringBuilder(
+                    "RegistryStore missing id. Unable to complete ");
+            if (!publish) {
+                logMessage.append("un");
             }
-        } catch (Exception e) {
-            LOGGER.error("Failed to publish registry configuration to {}",
-                    ((RegistryStoreImpl) registryStore).getRemoteName());
-        }
-    }
-
-    void registryUnpublish(RegistryStore registryStore) {
-
-        if (registryStore.getRegistryId()
-                .isEmpty()) {
+            logMessage.append("publish request.");
+            LOGGER.warn(logMessage.toString());
             return;
         }
-        try {
 
-            Optional<Metacard> registryIdentityMetacardOpt =
-                    Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
-
-            if (registryIdentityMetacardOpt.isPresent()) {
-                Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
-                String localRegistryId = registryIdentityMetacard.getAttribute(
-                        RegistryObjectMetacardType.REGISTRY_ID)
-                        .getValue()
-                        .toString();
-                Security.runAsAdminWithException(() -> {
-                    registryPublicationService.unpublish(localRegistryId,
-                            registryStore.getRegistryId());
-                    return null;
-                });
-            }
-        } catch (Exception e) {
-            LOGGER.error("Failed to unpublish registry configuration to ",
-                    ((RegistryStoreImpl) registryStore).getRemoteName());
-        }
+        executor.schedule(() -> {
+            try {
+                Optional<Metacard> registryIdentityMetacardOpt =
+                        Security.runAsAdminWithException(() -> federationAdminService.getLocalRegistryIdentityMetacard());
+
+                if (registryIdentityMetacardOpt.isPresent()) {
+                    Metacard registryIdentityMetacard = registryIdentityMetacardOpt.get();
+                    String localRegistryId = registryIdentityMetacard.getAttribute(
+                            RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString();
+                    Security.runAsAdminWithException(() -> {
+                        if (publish) {
```

Comment:
You could probably make it work with a `Consumer` instead of a Runnable. Something like `(foo) -> {...}`. That should make it easy enough to pass your other derived variable into the lambda.